sql/src/main/java/io/crate/jobs/JobExecutionContext.java

```diff
@@ -201,31 +199,28 @@ public long kill() {
             }
         }
 
+        // synchronous wait for all contexts to be killed
+        // TODO: make this method async
         try {
-            // synchronous wait for all contexts to be killed
             finishedFuture.get();
-            int currentNumSubContexts = numSubContexts.get();
-            assert currentNumSubContexts == 0 : "unexpected subContexts there: " + currentNumSubContexts;
-        } catch (Exception e) {
-            throw Throwables.propagate(e);
-        }
+        } catch (Exception ignored) {
 
+        }
+        assert numSubContexts.get() == 0 : "unexpected subContexts there: " + numSubContexts.get();
```

Comment:
please keep the local variable. If there is a bug numSubContexts could change between the two get calls and this would be very confusing for whoever has to debug this